### PR TITLE
move stringToJsRegex to grafana/ui

### PR DESCRIPTION
**Pull request by ryantxu**
In #15842, I need to use `stringToJsRegex`, but that is still in `kbn`.  This adds a `stringUtils.ts` class and in the grafana/ui package.

What do we want to do about depreciation messages?

```
/** deprecated since 6.1, use grafana/ui */
kbn.stringToJsRegex = str => {
  console.warn('Migrate stringToJsRegex to grafana/ui, ');
  return stringToJsRegex(str);
};
```

### CODE REVIEW

File: public/app/core/utils/kbn.ts [hunk 1]
```diff
@@ -228,18 +228,10 @@ kbn.slugifyForUrl = str => {
...
-  }
-
-  const match = str.match(new RegExp('^/(.*?)/(g?i?m?y?)$'));
-
-  if (!match) {
-    throw new Error(`'${str}' is not a valid regular expression.`);
-  }
-
-  return new RegExp(match[1], match[2]);
+  console.warn('Migrate stringToJsRegex to grafana/ui, ');
```
Comment (dprokop) [inline public/app/core/utils/kbn.ts]: Cool! WDYT about introducing `method/functionDeprecationWarning` in `utils` to have one generic message for such cases? I have created sth like that some time ago for props https://github.com/grafana/grafana/blob/master/packages/grafana-ui/src/utils/propDeprecationWarning.ts

File: packages/grafana-ui/src/utils/index.ts [hunk 2]
```diff
@@ -2,4 +2,5 @@ export * from './processTimeSeries';
 export * from './valueFormats/valueFormats';
 export * from './colors';
 export * from './namedColorsPalette';
+export * from './stringUtils';
```
Comment (dprokop) [inline packages/grafana-ui/src/utils/index.ts]: `stringUtils` -> `string` maybe? To follow convention `/utils/[scope].ts`

**Review by dprokop — APPROVED** [review]

File: public/app/core/utils/kbn.ts (hunk 1 above)
Comment (ryantxu) [reply public/app/core/utils/kbn.ts]: rather than add a new identical function, I renamed that `deprecationWarning.ts` and used it in both places.

Comment (ryantxu) [reply public/app/core/utils/kbn.ts]: If that sounds OK, i think this is ready to merge

**Pull request merged**